src/strings/nfa-strings.cc

```diff
@@ -144,6 +144,17 @@ void ShortestWordsMap::update_current_words(LengthWordsPair& act, const LengthWo
     act.first = dst.first + 1;
 }
 
+std::set<mata::Symbol> mata::strings::get_one_symbol_words(const Nfa& nfa) {
+    std::set<mata::Symbol> one_symbol_words;
+    for (State init : nfa.initial) {
+        for (const auto& move_from_init : nfa.delta[init]) {
+            if (!one_symbol_words.contains(move_from_init.symbol) && nfa.final.intersects_with(move_from_init.targets)) {
+                one_symbol_words.insert(move_from_init.symbol);
```

Comment:
No need to check for the symbol in `one_symbol_words` twice. Just check the intersection and try to insert in the true branch. If the key already exists, it will do nothing. `one_symbol_words` set will be searched only once.

Reply:
It is a bit of optimization, I do not want to do the intersection if not needed (I assume `contains` is much faster). But it is very possible it becomes slower with this "optimization".

Reply:
I do not think inserting to `std::set` with iterator pointing to its `.end()` makes sense.

Reply:
It does. It always inserts in front of the passed iterator. Therefore, if you pass end iterator to the method, it will essentially do `push_back()`.
```cpp
std::set<int> vs;
    vs.insert(vs.end(), 3);
    for (auto v : vs) {
        std::cout << v << " "; // Prints '3'.
    }
    std::cout << "\n";
```

Reply:
Ok, I added it.